packages/react-core/src/components/NumberInput/__tests__/NumberInput.test.tsx

```diff
@@ -60,4 +61,128 @@ describe('numberInput', () => {
+    userEvent.type(input, '55');
+
+    expect(onChangeMock).toHaveBeenCalledTimes(2);
+  });
+
+  test('does not call onBlur callback when input does not lose focus', () => {
+    const onBlurMock = jest.fn();
+
+    render(<NumberInput onBlur={onBlurMock}>5</NumberInput>);
+
+    const input = screen.getByRole('spinbutton');
+    userEvent.click(input);
+
+    expect(onBlurMock).not.toHaveBeenCalled();
+  });
+
+  test('calls onBlur callback when input loses focus', () => {
+    const onBlurMock = jest.fn();
+
+    render(<NumberInput onBlur={onBlurMock}>5</NumberInput>);
+
+    const input = screen.getByRole('spinbutton');
+    userEvent.click(input);
+    userEvent.click(document.body);
+
+    expect(onBlurMock).toHaveBeenCalledTimes(1);
+  });
+
+  test('removes trailing zero', () => {
+    render(<NumberInput value={120.0} />);
+    expect(screen.getByDisplayValue('120')).toBeInTheDocument();
+  });
+
+  test('removes many trailing zeroes', () => {
+    render(<NumberInput value={120.0100} />);
+    expect(screen.getByDisplayValue('120.01')).toBeInTheDocument();
+  });
+
+  test('removes leading zero', () => {
+    render(<NumberInput value={10} onChange={() => {}} />);
+  
+    const input = screen.getByRole('spinbutton');
+    userEvent.type(input, '{arrowleft}{arrowleft}0');
+    expect(input).toHaveDisplayValue('010');
+  
+    userEvent.click(document.body);
+  
+    expect(input).toHaveDisplayValue('10');
+  });
+
+  test('removes a trailing zero from a negative number', () => {
+    render(<NumberInput value={-40.0} />);
+    expect(screen.getByDisplayValue('-40')).toBeInTheDocument();
+  });
+
+  test('removes trailing zeroes from a negative number', () => {
+    render(<NumberInput value={-40.0100} />);
+    expect(screen.getByDisplayValue('-40.01')).toBeInTheDocument();
+  });
+
+  test('removes leading zero from a negative number', () => {
+    render(<NumberInput value={-18} onChange={() => {}} />);
+  
+    const input = screen.getByRole('spinbutton');
+    userEvent.type(input, '{arrowleft}{arrowleft}0');
+    expect(input).toHaveDisplayValue('-018');
+  
+    userEvent.click(document.body);
+  
+    expect(input).toHaveDisplayValue('-18');
+  });
+
+  test('removes leading and trailing zeroes from a positive number', () => {
+    render(<NumberInput value={47.0} onChange={() => {}} />);
+  
+    const input = screen.getByRole('spinbutton');
+    userEvent.type(input, '{arrowleft}{arrowleft}0');
+    expect(input).toHaveDisplayValue('047');
+  
+    userEvent.click(document.body);
+  
+    expect(input).toHaveDisplayValue('47');
+  });
+
+  test('removes leading and trailing zeroes from a negative number', () => {
+    render(<NumberInput value={-55.0} onChange={() => {}} />);
+  
+    const input = screen.getByRole('spinbutton');
+    userEvent.type(input, '{arrowleft}{arrowleft}0');
+    expect(input).toHaveDisplayValue('-055');
+  
+    userEvent.click(document.body)
+  
+    expect(input).toHaveDisplayValue('-55');
+  });
```

Comment:
I don't believe these tests are needed either since the leading zero removal behavior is already covered in the preceding tests.